SubterfugeCore/Core/Generation/MapGenerator.cs

```diff
@@ -109,18 +128,18 @@ public List<Outpost> GeneratePlayerOutposts()
             
             // setup variables
             double direction;
-            int distance;
+            float distance;
             bool usableLocation = true;
             int x, y, idx;
             RftVector vectorDistance;
             RftVector currentOutpostPosition;
             Outpost currentOutpost, otherOutpost;
 
             // Loop to generate outposts until the number of generated outposts is valid
-            while (playerOutposts.Count < this.OutpostsPerPlayer + this.DormantsPerPlayer)
+            while (playerOutposts.Count < this.Configuration.OutpostsPerPlayer + this.Configuration.DormantsPerPlayer)
             {
                 // calculate the new outposts location within allowable raidius
-                distance = this.RandomGenerator.NextRand(MinOutpostDistance, MaxSeedDistance);
+                distance = (float)(this.RandomGenerator.NextDouble() * (this.Configuration.MaxiumumOutpostDistance - this.Configuration.MinimumOutpostDistance)) + this.Configuration.MinimumOutpostDistance;
```

Comment:
Out of curiosity: why do we need a `float` here? I see that we're dealing with a `double` and some `int`s, I assume that manipulating a `double` with `int`s yields a new `double`, so why not just make the `distance` variable a `double` instead of a `float`?